src/WebJobs.Extensions.DurableTask/LocalHttpListener.cs

```diff
@@ -66,8 +55,35 @@ public async Task StartAsync()
             }
 
 #if !FUNCTIONS_V1
-            await this.localWebHost.StartAsync();
-            this.IsListening = true;
+            int numAttempts = 1;
+            do
+            {
+                int availablePort = this.GetAvailablePort();
+                try
+                {
+                    this.InternalRpcUri = new Uri($"http://127.0.0.1:{availablePort}/durabletask/");
+                    var listenUri = new Uri(this.InternalRpcUri.GetLeftPart(UriPartial.Authority));
+                    this.localWebHost = new WebHostBuilder()
+                        .UseKestrel()
+                        .UseUrls(listenUri.OriginalString)
+                        .Configure(a => a.Run(this.HandleRequestAsync))
+                        .Build();
+
+                    await this.localWebHost.StartAsync();
+                    this.IsListening = true;
+                    break;
+                }
+                catch
+                {
+                    this.traceHelper.ExtensionWarningEvent(
+                        this.durableTaskOptions.HubName,
+                        functionName: string.Empty,
+                        instanceId: string.Empty,
+                        message: $"Failed to open local socket {availablePort}. This was attempt #{numAttempts} to open a local port.");
+                    numAttempts++;
+                }
+            }
+            while (numAttempts <= 3);
```

Comment:
I wonder if we should try more times, like 10. I also wonder if we should have a small sleep in-between retries. I'm not sure how random the port selection is, so it would be nice if we could mitigate the case where two hosts are starting up at the same time and selecting the same random sequence of available port numbers.

Reply:
Makes sense. if we are worried about them being in sync, would doing a semi-random amount of sleep time help (i.e. between 50-100 ms)? That way we keep the two hosts out of lockstep.

Reply:
Added a dynamic sleep between 0 and 1 seconds here. Let me know if that's what you had in mind.